### PR TITLE
CirrusD should not kick members

### PR DESCRIPTION
**Pull request by fassadlr**
@quantumagi the DI approach was too complicated and will require a lot of refactoring. This will work for now.

### CODE REVIEW

File: src/Stratis.Bitcoin/Utilities/BinarySearch.cs [hunk 1]
```diff
@@ -17,8 +17,10 @@ private static T BinaryFindFirst<T>(T[] array, Func<T, bool?> func, int first, i
 
             // Otherwise split the array in two and search each half.
             int pivot = length / 2;
-            return BinaryFindFirst(array, func, first, pivot)
-                ?? BinaryFindFirst(array, func, first + pivot, length - pivot);
+            var result = BinaryFindFirst(array, func, first, pivot);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/Utilities/BinarySearch.cs:20]: @quantumagi the solution didnt build this my side. It was due to a coalescing feature only available in C#8

**Review by zeptin — APPROVED** [review]

File: src/Stratis.Bitcoin.Features.PoA/Voting/IdleFederationMembersKicker.cs [hunk 2]
```diff
@@ -119,14 +119,18 @@ public bool ShouldBeKicked(PubKey pubKey, uint blockTime, out uint inactiveForSe
...
+
             inactiveForSeconds = blockTime - lastActiveTime;
 
             return (inactiveForSeconds > this.federationMemberMaxIdleTimeSeconds && !this.federationManager.IsMultisigMember(pubKey));
         }
 
         private void OnBlockConnected(BlockConnected blockConnectedData)
         {
+            // Do nothing if this is not a mining node on Cirrus.
+            if (this.federationManager is FederationManager)
```
Comment (quantumagi) [inline src/Stratis.Bitcoin.Features.PoA/Voting/IdleFederationMembersKicker.cs:131]: Perhaps a negative test would be better: e.g. `if (!(this.federationManager is CollateralFederationManager))` ?

**Comment by quantumagi** [comment]
Something like this would be better:
```
        public static async Task MainAsync(string[] args)
        {
            try
            {
                var nodeSettings = new NodeSettings(networksSelector: CirrusNetwork.NetworksSelector,
                    protocolVersion: ProtocolVersion.CIRRUS_VERSION, args: args)
                {
                    MinProtocolVersion = ProtocolVersion.ALT_PROTOCOL_VERSION
                };

                ((PoAConsensusOptions)nodeSettings.Network.Consensus.Options).AutoKickIdleMembers = false;
```

**Review by quantumagi — CHANGES_REQUESTED** [review]
See comment.

**Comment by fassadlr** [comment]
Cool will fix in the next hour :)

**Comment by fassadlr** [comment]
@quantumagi done and tested 👍

**Review by quantumagi — APPROVED** [review]

**Pull request merged**